packages/angular_devkit/build_angular/src/utils/process-bundle.ts

```diff
@@ -0,0 +1,179 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import * as fs from 'fs';
+import * as path from 'path';
+import { SourceMapConsumer, SourceMapGenerator } from 'source-map';
+import { minify } from 'terser';
+
+const { transformAsync } = require('@babel/core');
+
+interface ProcessBundleOptions {
+  filename: string;
+  code: string;
+  map?: string;
+  sourceMaps: boolean;
+  runtime: boolean;
+  optimize: boolean;
+  optimizeOnly?: boolean;
+}
+
+export function process(
+  options: ProcessBundleOptions,
+  callback: (error: Error | null, result?: {}) => void,
+): void {
+  processWorker(options).then(() => callback(null, {}), error => callback(error));
+}
+
+async function processWorker(options: ProcessBundleOptions): Promise<void> {
+  // If no downlevelling required than just mangle code and return
+  if (options.optimizeOnly) {
+    return mangleOriginal(options);
+  }
+
+  // if code size is larger than 500kB, manually handle sourcemaps with newer source-map package.
+  // babel currently uses an older version that still supports sync calls
+  const codeSize = Buffer.byteLength(options.code, 'utf8');
+  const manualSourceMaps = codeSize >= 500 * 1024;
+
+  // downlevel the bundle
+  let { code, map } = await transformAsync(options.code, {
+    filename: options.filename,
+    inputSourceMap: !manualSourceMaps && options.map !== undefined && JSON.parse(options.map),
+    babelrc: false,
+    // modules aren't needed since the bundles use webpacks custom module loading
+    // loose generates more ES5-like code but does not strictly adhere to the ES2015 spec (Typescript is loose)
+    // 'transform-typeof-symbol' generates slower code
+    presets: [
+      ['@babel/preset-env', { modules: false, loose: true, exclude: ['transform-typeof-symbol'] }],
+    ],
+    minified: true,
+    sourceMaps: options.sourceMaps,
+  });
+
+  const newFilePath = options.filename.replace('es2015', 'es5');
+
+  // Adjust lazy loaded scripts to point to the proper variant
+  if (options.runtime) {
+    code = code.replace('"-es2015.', '   "-es5.');
```

Comment:
Shouldn't this be a global replace, and not replace only the first occurrence? 

Nit: remove the leading  spaces prior to `"-es5.`

Reply:
The spaces keeps the line positions intact for source map purposes.

Reply:
Maybe an inline comment would help :)

Reply:
Done.